lib/redux/middleware/discover_middleware.dart

```diff
@@ -1,17 +1,17 @@
+import 'package:farmsmart_flutter/data/repositories/articles_directory_repository.dart';
 import 'package:farmsmart_flutter/redux/app/app_state.dart';
-import 'package:farmsmart_flutter/data/repositories/articles_repository.dart';
 import 'package:farmsmart_flutter/redux/home/discover/discover_actions.dart';
 import 'package:redux/redux.dart';
 
 class DiscoverMiddleWare extends MiddlewareClass<AppState> {
   @override
   Future call(Store<AppState> store, dynamic action, NextDispatcher next) async {
-    if(action is FetchArticleListAction) {
-        var listOfArticles = await ArticlesRepository.get().getListOfArticles();
-        listOfArticles = await ArticlesRepository.get().getListOfArticlesWithImages(listOfArticles);
-        store.dispatch(UpdateArticleListAction(listOfArticles));
+    if(action is FetchArticleDirectoryAction) {
+        var articlesDirectory = await ArticlesDirectoryRepository.get().getDirectory();
+        articlesDirectory = await ArticlesDirectoryRepository.get().getListOfArticles(articlesDirectory);
```

Comment:
not sure I understand this function, I feel there should be a single use case here and not accumulating in the get. let´s have a tech scrum over this.

Reply:
This function is now modified. @albertmiro help us to improve it.